apps/api/src/app/integrations/e2e/create-integration.e2e.ts

```diff
@@ -95,6 +95,52 @@ describe('Create Integration - /integration (POST)', function () {
     expect(firstIntegration.active).to.equal(false);
     expect(secondIntegration.active).to.equal(true);
   });
+
+  it('should create custom SMTP integration with TLS options successfully', async function () {
+    const payload = {
+      providerId: 'nodemailer',
+      channel: 'email',
+      credentials: {
+        host: 'smtp.example.com',
+        port: '587',
+        secure: 'true',
+        requireTls: true,
+        tlsOptions: { rejectUnauthorized: false },
+      },
+      active: true,
+      check: false,
+    };
+
+    const environmentId = (await session.testAgent.get(`/v1/integrations`)).body.data[0]._environmentId;
+
+    await session.testAgent.post('/v1/integrations').send(payload);
+
+    const integrations = await integrationRepository.findByEnvironmentId(environmentId);
+
+    const sendGridIntegration = integrations.find((i) => i.providerId.toString() === 'sendgrid');
+    const nodeMailerIntegration = integrations.find((i) => i.providerId.toString() === 'nodemailer');
+
+    expect(sendGridIntegration?.active).to.equal(false);
+    expect(nodeMailerIntegration?.credentials?.host).to.equal(payload.credentials.host);
+    expect(nodeMailerIntegration?.credentials?.port).to.equal(payload.credentials.port);
+    expect(nodeMailerIntegration?.credentials?.secure).to.equal(payload.credentials.secure);
+    expect(nodeMailerIntegration?.credentials?.requireTls).to.equal(payload.credentials.requireTls);
+    expect(nodeMailerIntegration?.credentials?.tlsOptions).to.instanceOf(Object);
+    expect(nodeMailerIntegration?.credentials?.tlsOptions).to.eql(payload.credentials.tlsOptions);
+    expect(nodeMailerIntegration?.active).to.equal(true);
+
+    //database checks
+    const nodeMailerStoredIntegration = (await integrationRepository.findByEnvironmentId(environmentId)).find(
+      (i) => i.providerId.toString() === 'nodemailer'
+    );
+    expect(nodeMailerStoredIntegration?.credentials?.host).to.equal(payload.credentials.host);
+    expect(nodeMailerStoredIntegration?.credentials?.port).to.equal(payload.credentials.port);
+    expect(nodeMailerStoredIntegration?.credentials?.secure).to.equal(payload.credentials.secure);
+    expect(nodeMailerStoredIntegration?.credentials?.requireTls).to.equal(payload.credentials.requireTls);
+    expect(nodeMailerStoredIntegration?.credentials?.tlsOptions).to.instanceOf(Object);
+    expect(nodeMailerStoredIntegration?.credentials?.tlsOptions).to.eql(payload.credentials.tlsOptions);
+    expect(nodeMailerStoredIntegration?.active).to.equal(true);
```

Comment:
All these checks are already done between lines 121 and 130. Not need to do them again. 🙂